src/services/EscalationLogic.ts

```diff
@@ -0,0 +1,70 @@
+import MailService from '../services/MailService';
+import PushService from '../services/PushService';
+import TemplatingService from '../services/TemplatingService';
+import MessageModel from '../models/message';
+import Message from '@/interfaces/Message';
+
+export default class EscalationLogic {
+  // region public static methods
+  // endregion
+
+  // region private static methods
+  // endregion
+
+  // region public members
+  // endregion
+
+  // region private members
+
+  private mailService: MailService;
+  private pushService: PushService;
+
+  // endregion
+
+  // region constructor
+
+  public constructor() {
+    this.mailService = new MailService();
+    this.pushService = new PushService();
+  }
+
+  // endregion
+
+  // region public methods
+
+  public escalate(messageId: string) {
+    MessageModel.findById(messageId, (err, message: Message) => {
```

Comment:
Use `await` instead of providing a callback to reduce the nesting level. FindById returns a Promise, when not provided a callback.

Reply:
If we do not use the callback then the type of the returned object is `IMessageModel` and not `Message`. Since the interface for `IMessageModel` looks different than the one for `Message`, we can not access all properties. Do you know how to solve this?

Reply:
I refactored the escalation logic. In order to avoid the callback, I extended the IMessageModel interface with the needed properties. Please review `src/models/message.ts` and `src/services/EscalationLogic.ts`.